routing/localchans/manager.go

```diff
@@ -212,6 +212,14 @@ func (r *Manager) updateEdge(tx kvdb.RTx, chanPoint wire.OutPoint,
 		edge.MinHTLC = *newSchema.MinHTLC
 	}
 
+	// If the channel is unannounced we make sure the dont_forward bit is
+	// set in the ChannelEdgePolicy. This is an on-the-fly migration. The
+	// edge is persisted to disk later when propagating the new channel
+	// update.
+	if edgeInfo.AuthProof == nil {
+		edge.MessageFlags |= lnwire.ChanUpdateDontForward
```

Comment:
should instead be done here: https://github.com/lightningnetwork/lnd/blob/13568fd5b09435f8254a91527996091bef30996e/discovery/gossiper.go#L1729

if `updatechanpolicy` is called while the channel is usable but before 6 confs (say min depth has been reached), then the flag will be set for the channel indefinitely

Reply:
I think we always persist the new state after the six conf ChanUpdate is sent out?
https://github.com/lightningnetwork/lnd/blob/13568fd5b09435f8254a91527996091bef30996e/discovery/gossiper.go#L2899-L2920

Apart from that, I think I had a bug here, I was also setting the dont_forward bit in the time period between the six-blocks and our required conf target (default 3 blocks). I tested with the latest eclair release and they do not set it for public channel in this phase, so I made sure now that we fetch the current channel infos so that we do only set the dont_forward bit in the cases you mentioned above:

- the channel doesn't have the announce_channel bit set
- channel uses aliases, which is also the case for public zeroconf channel before the conf target of 6

Reply:
> I think we always persist the new state after the six conf ChanUpdate is sent out?

that's true if we're deleting the edge and re-adding it. but if scid-alias isn't applied, i think we don't resend the edge to the gossiper.

Reply:
After consulting with the other Implementations we decided in the spec meeting https://github.com/lightning/bolts/issues/1107 to set the dont_forward bit for the initial ChannelUpdate if the maturity of the channel is below 6 confirmations. Therefore I had to add an additional announcement of the ChannelUpdate after 6 blocks for public channels. Although for channels which have an active Alias this ChanUpdate is now redundant  what do you think ?
Moreover I had to add a check that we do not broadcast ChannelUpdates which we fetch from the db although the dont_forward bit is still set, so that we only broadcast ChannelUpdates if the dont_forward bit is not set.
I decided against your proposal to move the setting of the MsgFlags to the gossiper because we would have to resign these ChanUpdates and also would never persist the ChannelUpdate for private channels to disk (at least for channels prior to this PR), happy to reconsider it after I heard your thoughts.